### DISCUSSION TITLE
How do I copy text with the right mouse button?

### DISCUSSION THREAD

**Discussion by Vaman93** [sourcelair/xterm.js#4308] [Q&A] [answered]
How to enable copy select text mouse right click?

**Comment by Tyriar** [comment, marked as answer]
Here's how VS Code does it:

https://github.com/microsoft/vscode/blob/e30f8b58daf2a174787ae436c0d5dffda35d6981/src/vs/workbench/contrib/terminal/browser/terminalTabbedView.ts#L354-L381

https://github.com/microsoft/vscode/blob/e30f8b58daf2a174787ae436c0d5dffda35d6981/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1129-L1150

Basically, listen to `mousedown` on the container that xterm was opened on and handle the copy manually via `Terminal.getSelection()` and the `navigator.clipboard` API.

**Reply by Vaman93** [reply]
Thank you @Tyriar
